File: 731-my-calendar-ii/731-my-calendar-ii.java
class MyCalendarTwo {
    
    TreeMap<Integer,Integer> map;
    public MyCalendarTwo() {
        this.map = new TreeMap<>();
    }
    
    public boolean book(int start, int end) {
        map.put(start,map.getOrDefault(start,0) + 1);
        map.put(end  ,map.getOrDefault(end  ,0) - 1);
        
        int sum = 0;
        for(var entry : map.entrySet()){
            sum+=entry.getValue();
            if(sum>2){
                map.put(start,map.getOrDefault(start,0) - 1);
                map.put(end  ,map.getOrDefault(end  ,0) + 1);
                return false;
            }
        }
        return true;
    }
}

/**
 * Your MyCalendarTwo object will be instantiated and called as such:
 * MyCalendarTwo obj = new MyCalendarTwo();
 * boolean param_1 = obj.book(start,end);
 */

File: 731-my-calendar-ii/731-my-calendar-ii.cpp
class MyCalendarTwo {
public:
    map<int,int> m;
    MyCalendarTwo() {
        
    }
    
    bool book(int start, int end) {
        int sum=0;
        m[start]++;
        m[end]--;
        for(auto x:m){
            sum+=x.second;
            if(sum>2){
                m[start]--;
                m[end]++;
                return false;
            }
        }
        return true;
    }
};